quesma/queryparser/pancake_sql_query_generation.go

```diff
@@ -214,6 +247,26 @@ func (p *pancakeSqlQueryGenerator) generateSelectCommand(aggregation *pancakeMod
 		}
 
 		if layer.nextBucketAggregation != nil {
+			if filter, isFilter := layer.nextBucketAggregation.queryType.(bucket_aggregations.FilterAgg); isFilter {
+
+				for i, newFilterColumn := range layer.nextBucketAggregation.selectedColumns {
+					aliasName := fmt.Sprintf("%s_col_%d", layer.nextBucketAggregation.internalName, i)
+					aliasedColumn := model.NewAliasedExpr(newFilterColumn, aliasName)
+					selectColumns = append(selectColumns, aliasedColumn)
+				}
+
+				if layerId+1 < len(aggregation.layers) {
+					addSelectColumns, err := p.generateLeafFilter(aggregation.layers[layerId+1], filter.WhereClause)
+					if err != nil {
+						return nil, false, err
+					}
+					selectColumns = append(selectColumns, addSelectColumns...)
+				}
+				break
```

Comment:
Is this break needed?

Reply:
Yes, but this keeps me puzzled how to refactor it better.

Currently the logic is that filters are leaf, but they might be one more level to process. I escape the standard logic.